app/viewmodels/repository.js

```diff
@@ -1,11 +1,18 @@
-'use strict';
+﻿'use strict';
 define(['plugins/http', 'durandal/app', 'knockout', '../modules/github'], function (http, app, ko, Github) {
-    var github = new Github();
+    var github = Github.instance;
+
+    // defining these in the function scope as linkToDocument is not
+    // called in the right context, and this is cleaner than a bind in
+    // the template
+    var repositoryOwner =  ko.observable('');
+    var repositoryName = ko.observable('');
+    var commitSha = ko.observable('');
```

Comment:
I understand keeping the template clean, but I'm not clear  what you mean by _is not called in the right context_.
Defining the observables inline with the returned object should have the same result as defining them here. 
I thought that you might prefer this approach because it avoids `this` in `linkToDocument`.